File: Array/array.py
class Array: 
    def __init__(self):
        self.length = 0;
        self.data = {};
    
    def get(self, index):
        return self.data[index];

    def push(self, item):
        self.data[self.length] = item;
        self.length += 1;
        return item;

    def pop(self):
        lastItem = self.data[self.length - 1];
        del self.data[self.length - 1];
        self.length -= 1;
        return lastItem;

    def delete(self, index):
        deletedItem = self.data[index];
        self.shiftItems(index)
        self.length -= 1;
        return deletedItem;

    def shiftItems(self, index):
        for i in range(index, self.length - 1):
            self.data[i] = self.data[i + 1];  
        del self.data[self.length - 1];
        
    
    def getData(self):
        return self.data;

myArr = Array();
myArr.push('foo')
myArr.push('foo2')
myArr.push('foo3')
myArr.push('foo4')
myArr.push('foo5')
myArr.push('foo6')
myArr.delete(1)
print(myArr.getData())

File: Array/array.js
class MyArray {
  constructor() {
    this.legth = 0;
    this.data = {};
  }

  get(index) {
    return this.data[index];
  }

  push(item) {
    this.data[this.legth] = item;
    this.legth++;
    return item;
  }

  pop() {
    const lastItem = this.data[this.legth - 1];
    delete this.data[this.legth - 1];
    this.legth--;
    return lastItem;
  }

  delete(index) {
    const deletedItem = this.data[index];
    this.shiftItems(index);
    return deletedItem;
  }

  shiftItems(index) {
    for (let i = index; i < this.legth - 1; i++) {
      this.data[index] = this.data[index + 1];
    }

    delete this.data[this.legth - 1];
    this.legth--;
  }
}

const MyArr = new MyArray();

MyArr.push('hd');
MyArr.push('food');
MyArr.push('food2');
MyArr.push('food3');
MyArr.push('food4');
MyArr.push('food5');
MyArr.push('food6');
MyArr.push('food6');
MyArr.push('food9');
MyArr.delete(5);
console.log(MyArr.pop());

console.log(MyArr);

// give me some bubble sort code
